app/backend/src/tests/test_account.py

```diff
@@ -40,6 +40,50 @@ def test_ChangePassword_normal(db, fast_passwords):
         assert updated_user.hashed_password == hash_password(new_password)
 
 
+def test_ChangePassword_normal_short_password(db, fast_passwords):
+    # user has old password and is changing to new password, but used short password
+    old_password = random_hex()
+    new_password = random_hex(length=1)
+    user, token = generate_user(hashed_password=hash_password(old_password))
+
+    with account_session(token) as account:
+        with pytest.raises(grpc.RpcError) as e:
+            account.ChangePassword(
+                account_pb2.ChangePasswordReq(
+                    old_password=wrappers_pb2.StringValue(value=old_password),
+                    new_password=wrappers_pb2.StringValue(value=new_password),
+                )
+            )
+        assert e.value.code() == grpc.StatusCode.INVALID_ARGUMENT
+        assert e.value.details() == errors.PASSWORD_TOO_SHORT
+
+    with session_scope() as session:
+        updated_user = session.query(User).filter(User.id == user.id).one()
+        assert updated_user.hashed_password == hash_password(old_password)
+
+
+def test_ChangePassword_normal_insecure_password(db, fast_passwords):
+    # user has old password and is changing to new password, but used insecure password
+    old_password = random_hex()
+    new_password = "12345678"
+    user, token = generate_user(hashed_password=hash_password(old_password))
+
+    with account_session(token) as account:
+        with pytest.raises(grpc.RpcError) as e:
+            account.ChangePassword(
+                account_pb2.ChangePasswordReq(
+                    old_password=wrappers_pb2.StringValue(value=old_password),
+                    new_password=wrappers_pb2.StringValue(value=new_password),
+                )
+            )
+        assert e.value.code() == grpc.StatusCode.INVALID_ARGUMENT
+        assert e.value.details() == errors.INSECURE_PASSWORD
+
+    with session_scope() as session:
+        updated_user = session.query(User).filter(User.id == user.id).one()
+        assert updated_user.hashed_password == hash_password(old_password)
+
+
```

Comment:
You could also test just the password complexity check function as it's quite modular from the rest of the code, but that's a bit overkill.